examples/nextjs-boilerplate/src/components/send-sol.tsx

```diff
@@ -0,0 +1,190 @@
+import React, { useEffect, useState } from 'react';
+import { useConnection, useWallet } from '@solana/wallet-adapter-react';
+import { CANVAS_WALLET_NAME } from '@dscvr-one/canvas-wallet-adapter';
+import {
+  PublicKey,
+  SystemProgram,
+  Transaction,
+  TransactionMessage,
+  VersionedTransaction,
+} from '@solana/web3.js';
+import { useCanvasContext } from '@/context/canvas';
+
+export default function SendSol() {
+  const canvasContext = useCanvasContext();
+  const walletContext = useWallet();
+  const connectionContext = useConnection();
+  const [successfulSignedTx, setSuccessfulSignedTx] = useState<string>();
+  const [errorMessage, setErrorMessage] = useState<string>();
+  const [targetAddress, setTargetAddress] = useState<string>('');
+  const [amount, setAmount] = useState<number>(0);
+
+  // since wallets list not only `DSCVR Canvas` but also other extensions, we do a default select to DSCVR Canvas.
+  useEffect(() => {
+    if (walletContext.wallet?.adapter.name === CANVAS_WALLET_NAME) return;
+    const exists = walletContext.wallets.find(
+      (w) => w.adapter.name === CANVAS_WALLET_NAME,
+    );
+    if (exists) {
+      walletContext.select(exists.adapter.name);
+    }
+  }, [walletContext, walletContext.wallets, walletContext.wallet]);
```

Comment:
Is `walletContext` sufficient enough here, not sure you need the others

Reply:
```suggestion
  }, [walletContext.wallet?.adapter.name, walletContext.wallets]);
```

Reply:
Thanks, I was trying to avoid a warning, turns our react uses this comment for those `// eslint-disable-next-line react-hooks/exhaustive-deps` 
So changed, thanks